src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -657,7 +657,7 @@ public void CopyTo(Array array, int index)
                     Debug.Assert(span.Length > 0);
                     Debug.Assert(m_array.Length > quotient);
                     // mask the final byte
-                    span[span.Length - 1] = (byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
+                    span[remainder] = (byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
```

Comment:
Do we need to zero bytes above remainder as well?  IOW if remainder is 2 should the  last byte be zeroed, the  3rd byte be masked, and first two copied?

Reply:
Nevermind I see this is a new array and will be zero’ed by GC.

Reply:
This is writing to an array that the user passed in. I don't think we'd want to zero the bytes outside the range we are copying into.